docs/proposals/gpu-support.md

```diff
@@ -0,0 +1,305 @@
+|---------------------|---------------------------------------------------------------------------------------------------|--------------------------------------------------------------|
+| Use of cgo          | Confined to binary                                                                                | Linked into kubelet, but with lazy binding                   |
+| Expandibility       | Limited if we run the plugin, increased if library is used to build a Kubernetes-tailored daemon. | Can reuse the `nvidia` library as we prefer                  |
+| Bloat               | None                                                                                              | Larger kubelet, even for systems without GPUs                |
+| Reliability         | Need to handle the binary disappearing at any time                                                | Fewer headeaches                                             |
+| (Un)Marshalling     | Need to talk over JSON                                                                            | None                                                         |
+| Administration cost | One more daemon to install, configure and monitor                                                 | No extra work required, other than perhaps configuring flags |
+| Releases            | Potentially on its own schedule                                                                   | Tied to Kubernetes'                                          |
+
+For all its drawbacks, the external binary solution is probably a better fit for
+V0.
+
+However we obtain the data, we'll need to change Kubelet's `dockertools` to also
+map devices into the containers and manage the CUDA volumes. As shown above,
+volumes are named after the hardware driver version.
+
+## Timeline
+
+### V0 (targeting Kubernetes 1.3)
+
+The first two tracks can progress in parallel.
+
+#### Scheduling
+
+1. Write scheduler extender, assuming one GPU per machine and user-supplied node
+labels for resource inventory
+1. Support fractional resources so multiple pods can share the same GPU
+
+#### Runtime
+
+1. Add kubelet config parameter to point to external resource plugin
+1. Teach kubelet to call plugins for pods requiring special resources, in
+order to
+    1. setup volumes (if needed) and obtain their names
+    1. map logical device to device node(s)
+    1. remove volumes
+1. Add Dockertools hook to map host devices
+1. Make kubelet map devices and volumes into containers
+
+#### Other
+
+1. Administrator documentation, with sample scripts
+1. User documentation
+
+### V1 (stretch goal for v1.3, v1.4+)
+
+- Teach kubelet to report resources discovered through plugins
+- Replace node labels with first-class resources
+- Support bare metal setups (mostly a matter of testing)
```

Comment:
Sorry for bothering.

Please talk about implementation details only. Track timeline in other ways.
It sounds like goals. If so, move it to "Objective" section.